lib/OBSWebSocket.js

```diff
@@ -20,50 +20,51 @@ class OBSWebSocket extends Socket {
   // Generates a messageId internally and will override any passed in the args.
   // Note that the requestType here is pre-marshaling and currently must match exactly what the websocket plugin is expecting.
   send(requestType, args = {}, callback) {
+    // TODO: Improve this to ensure args an object, not a function or primitive or something.
     args = args || {};
 
     return new Promise((resolve, reject) => {
-      if (!this._connected) {
-        debug('[send] %s', Status.NOT_CONNECTED.description);
-        this._doCallback(callback, Status.NOT_CONNECTED, null);
-        reject(Status.NOT_CONNECTED);
-        return;
-      }
+      const messageId = generateMessageId();
+      let rejectReason;
 
       if (!requestType) {
-        debug('[send] %s', Status.REQUEST_TYPE_NOT_SPECIFIED.description);
-        this._doCallback(callback, Status.REQUEST_TYPE_NOT_SPECIFIED, null);
-        reject(Status.REQUEST_TYPE_NOT_SPECIFIED);
-        return;
+        rejectReason = Status.REQUEST_TYPE_NOT_SPECIFIED;
```

Comment:
These should at least contain `status` and `error` to be in line with what the plugin spits out.